### PR TITLE
Add clang's python bindings

### PR DESCRIPTION
**Pull request by chrisburr**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages
- [x] Build number is 0
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

Adds clang's python bindings as a separate package. Once this is merged I'll add it for the other versions of `clangdev` that are supported by conda-forge.

### CODE REVIEW

**Comment by chrisburr** [comment]
@conda-forge/staged-recipes This is ready to review.

The Windows build failure can be ignored as this is a `noarch` package.

File: recipes/clang-bindings-python/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+source:
+  url: http://llvm.org/releases/{{ version }}/cfe-{{ version }}.src.tar.xz
+  sha256: 019f23c2192df793ac746595e94a403908749f8e0c484b403476d2611dd20970
+
+build:
+  number: 0
+  noarch: python
+  script:
+    - mkdir -p "${SP_DIR}"
+    - cp -r bindings/python/clang "${SP_DIR}"
```
Comment (ocefpaf) [inline recipes/clang-bindings-python/meta.yaml]: This is a re-package, right? Is it safe to make it noarch?
How hard would it be to build it from source?
(Sorry if the answers are obvious but I have no idea what packaging `clang-bindings-python` entails.)

Comment (chrisburr) [reply recipes/clang-bindings-python/meta.yaml]: No it's [effectively a single file python module](https://github.com/llvm-mirror/clang/tree/master/bindings/python/clang).

It's included in the clang source code and the [README states](https://github.com/llvm-mirror/clang/tree/master/bindings/python) to use it by setting `PYTHONPATH` but I think copying it to the site packages directory is cleaner.

Comment (chrisburr) [reply recipes/clang-bindings-python/meta.yaml]: The complicated bit is building `libclang` itself but that is handled by the `clangdev` feedstock.

**Comment by isuruf** [comment]
Why not create a new output from the same recipe?

**Comment by isuruf** [comment]
Also, how about `clang-python` or `python-clang` as the name?

**Comment by chrisburr** [comment]
Sure, I’ve changed the name to python-clang.

Is it possible to add it as an output to clangdev without having one build per Python version?

**Comment by isuruf** [comment]
Yes, see https://github.com/conda-forge/brial-feedstock/blob/master/recipe/meta.yaml

**Comment by chrisburr** [comment]
I started preparing the pull request to `clangdev` and remembered that it would also be useful to deploy this for `clang =4` so it can be installed in the same environment as the macOS compilers. I've tested it locally and this works provided the bindings are taken from clang 4 tarball.

As a result I'm inclined to keep it as an independent feedstock, what do you think?

**Comment by isuruf** [comment]
You could send a PR for the 4.x branch for clangdev. (And depend on clang=4)

**Comment by chrisburr** [comment]
[There isn't a `4.x` branch](https://github.com/conda-forge/clangdev-feedstock/branches). I'm not convinced it's worth the effort required to add a recipe for clang 4 and rebuild clang 5/7 (and maybe clang 6 for completeness).

File: recipes/clang-bindings-python/meta.yaml (hunk 1 above)
Comment (ocefpaf) [reply recipes/clang-bindings-python/meta.yaml]: > No it's effectively a single file python module .

Got it. Thanks!

**Comment by ocefpaf** [comment]
While I agree with @isuruf, a multiple output recipe would be better, this is a simple recipe that solves the problem "right now." Let's focus on the multiple outputs later and archive this feedstock when that is done.

**Pull request merged**